crates/matrix-sdk-base/src/read_receipts.rs

```diff
@@ -266,12 +266,21 @@ impl RoomReadReceipts {
 }
 
 /// Provider for timeline events prior to the current sync.
+#[cfg(not(target_arch = "wasm32"))]
 pub trait PreviousEventsProvider: Send + Sync {
     /// Returns the list of known timeline events, in sync order, for the given
     /// room.
     fn for_room(&self, room_id: &RoomId) -> Vector<SyncTimelineEvent>;
 }
 
+/// Provider for timeline events prior to the current sync.
+#[cfg(target_arch = "wasm32")]
+pub trait PreviousEventsProvider {
+    /// Returns the list of known timeline events, in sync order, for the given
+    /// room.
+    fn for_room(&self, room_id: &RoomId) -> Vector<SyncTimelineEvent>;
+}
```

Comment:
No need to duplicate the trait definition, you can update the supertraits from `Send + Sync` to `SendOutsideWasm + SyncOutsideWasm`.

Reply:
Fixed.